duckduckgo/duckduckgo-privacy-extension
# More https upgrades

**Reviewer:**



## Description:

Https upgrading from a bloom filter. 

## Steps to test this PR:
1. Open up the extension console and explicitly go to a popular site using http. You should see a line in the console that says `HTTPS: upgrade request url <your site>`.
2. Double check the URL was upgraded and that the popup says "Encrypted Connection".
3. You can test list updates by changing the alarm time and messing with the etags. https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/235/files#diff-3cd94a6f60c9bfd33dffeca9dc0735f8R191
4. You can test corrupt data by clearing indexedDB or by forcing the checksum function to return false. In this case the HTTPSStorage.getLists() function should completely fail. 
https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/235/files#diff-b3d1f859c76ee3ad6e55e5ccd6500bbaR91
5. Problems with HTTPS.setLists() should cause the isReady flag to set to false. You can test this by causing errors in setLists.  https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/235/files#diff-8d433432227571fe75ece3d6c6f8b018R18

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
